sdk/keyvault/Azure.Security.KeyVault.Shared/src/KeyVaultPipeline.cs

```diff
@@ -88,17 +90,28 @@ public Request CreateRequest(RequestMethod method, params string[] path)
 
             request.Uri.AppendQuery("api-version", ApiVersion);
 
-            return request;
+            return message;
         }
 
-#pragma warning disable CA1822 // Member can be static
-        public Response<T> CreateResponse<T>(Response response, T result)
+        public static NullableResponse<T> CreateResponse<T>(HttpMessage message, Func<T> resultFactory)
+            where T : IJsonDeserializable
+        {
+            if (message.Response.Status == 200 || message.Response.Status == 201)
```

Comment:
Do we know this is the full set of possible status codes for all cases?  Is there another check (like whether the response.Content is non-null) that might be valid here and reach a broader set of cases?  I defer to you on service logic, so if not, you can disregard this comment.

Reply:
I agree this is a hard call, but I don't want to check `Content`. The idea is that `Content` was non-null but had only an `{"error":{}}` object such that our factory pattern for deserializing would instantiate an "empty" object, such that `Response<KeyVaultSecret>.Value` was non-null but otherwise useless. I would want either a null `Value` or throw, such as this implementation does consistent with its documentation for `Value`.

Going by the success codes further down we supported before - 200, 201, 202, and 204 - 200 and 201 should really be the only values that should return content. Well, maybe 202. Just to be sure, I'll add that in. We would not expect 204 to return content, given it's `204 No Content`.